src/nvim/insexpand.c

```diff
@@ -1289,6 +1289,19 @@ void ins_compl_show_pum(void)
   }
 }
 
+void compl_set_info(char *info, const char *pum_text)
```

Comment:
Hmm, using `pum_text` as the key is a little weird, but might be best for now as `comp` is a linked list. If it was an array, the index should be same as the index into `pum_array[i]`..  

perhaps could add a doc comment with a `TODO` like "TODO: If compl_first_match is refactored to be an array, this could take the index"

Reply:
yes. plan to use VirtText refactor there after completepopup done.